README.md

```diff
@@ -62,6 +62,7 @@ This constructor is used to transform an [API request](http://kuzzle.io/api-refe
 | `connectionId` | `string` | Passed to [RequestContext](#modelsrequestcontext) constructor |
 | `error` | `KuzzleError` or `Error` | Invokes [setError](#seterrorerror) at initialization |
 | `protocol` | `string` | Passed to [RequestContext](#modelsrequestcontext) constructor |
+| `headers` | `object` | Passed to [RequestContext](#modelsrequestcontext) constructor |
```

Comment:
1. This proposal is a no-go, as there will be conflicts if there is a `headers` optional parameter in a client API route, which might happen considering this is a pretty common term
2. Can be considered
3. Same remark as for the 1st proposal

IMO, either the second option should be considered, or this fourth one, similar to your second proposal without the need to change the constructor of `RequestInput`:

```js
class Request {
  constructor(data, options) {
    this[_input] = new RequestInput(data);

    if (options && options.headers) {
      this[_input].setHeaders(options.headers);
      delete options.headers;
    }

    this[_context] = new RequestContext(options);
  }
}
```